### PR TITLE
small fix for loading without attrs

### PR DESCRIPTION
**Pull request by antonydellavecchia**
can load without attributes if desired

### CODE REVIEW

File: src/Serialization/serializers.jl [hunk 1]
```diff
@@ -143,7 +143,7 @@ mutable struct DeserializerState
   obj::Union{Dict{Symbol, Any}, Vector, JSON3.Object, JSON3.Array, BasicTypeUnion}
   key::Union{Symbol, Int, Nothing}
   refs::Union{Dict{Symbol, Any}, JSON3.Object, Nothing}
-  type_attr_map::Dict{String, Vector{Symbol}}
+  type_attr_map::Bool
```
Comment (antonydellavecchia) [inline src/Serialization/serializers.jl]:
```suggestion
  type_attr::Bool
```

File: src/Serialization/serializers.jl [hunk 2]
```diff
@@ -143,7 +143,7 @@ mutable struct DeserializerState
   obj::Union{Dict{Symbol, Any}, Vector, JSON3.Object, JSON3.Array, BasicTypeUnion}
   key::Union{Symbol, Int, Nothing}
   refs::Union{Dict{Symbol, Any}, JSON3.Object, Nothing}
-  type_attr_map::Dict{String, Vector{Symbol}}
+  type_attr::Bool
```
Comment (antonydellavecchia) [inline src/Serialization/serializers.jl]:
```suggestion
  with_attrs::Bool
```

**Comment by lgoettgens** [comment]
The changes look great!
I think it would be great if we could have some dedicated accessor to `DeserializerState.with_attrs`. Furthermore, I would then like it if we would have the same accessor for `SerializerState` as well. This would enable the following refactoring of `SerializerState`: Don't save the `type_attr_map` in `SerializerState`, instead just `with_attrs::Bool`. The only place where this then needs a change is `attrs_list`, which would return `get(Oscar.type_attr_map, encode_type(T), Symbol[])` if `s.with_attrs` and `Symbol[]` otherwise. So there would be no need to pass the type_attr_map dict down through all the functions. What do you think about this?

File: src/Serialization/serializers.jl [hunk 3]
```diff
@@ -251,7 +251,7 @@ end
 
 const state_types = Union{SerializerState, DeserializerState}
```
Comment (lgoettgens) [inline src/Serialization/serializers.jl]:
```suggestion
```
This union is no longer used anywhere

**Comment by antonydellavecchia** [comment]
> The changes look great! I think it would be great if we could have some dedicated accessor to `DeserializerState.with_attrs`. Furthermore, I would then like it if we would have the same accessor for `SerializerState` as well. This would enable the following refactoring of `SerializerState`: Don't save the `type_attr_map` in `SerializerState`, instead just `with_attrs::Bool`. The only place where this then needs a change is `attrs_list`, which would return `get(Oscar.type_attr_map, encode_type(T), Symbol[])` if `s.with_attrs` and `Symbol[]` otherwise. So there would be no need to pass the type_attr_map dict down through all the functions. What do you think about this?

the reason it was like this, was to at some point in the future allow overrides from the user, possible something to discuss at the meeting in kaiserslautern

**Comment by antonydellavecchia** [comment]
> > The changes look great! I think it would be great if we could have some dedicated accessor to `DeserializerState.with_attrs`. Furthermore, I would then like it if we would have the same accessor for `SerializerState` as well. This would enable the following refactoring of `SerializerState`: Don't save the `type_attr_map` in `SerializerState`, instead just `with_attrs::Bool`. The only place where this then needs a change is `attrs_list`, which would return `get(Oscar.type_attr_map, encode_type(T), Symbol[])` if `s.with_attrs` and `Symbol[]` otherwise. So there would be no need to pass the type_attr_map dict down through all the functions. What do you think about this?
> 
> the reason it was like this, was to at some point in the future allow overrides from the user, possible something to discuss at the meeting in kaiserslautern.

I could add an access-er but this may change in the future. For example if the user would like to have some attributes but not all, then we will go back to some attribute map

**Review by lgoettgens — APPROVED** [review]
This is already an improvement to current master, as it brings the functionality to the semantics described in the docstring. Let's postpone my other points to the meeting in KL

**Pull request merged**